src/MatrixClientPeg.ts

```diff
@@ -261,7 +261,7 @@ class _MatrixClientPeg implements IMatrixClientPeg {
     }
 
     public getHomeserverName(): string {
-        const matches = /^@.+:(.+)$/.exec(this.matrixClient.credentials.userId);
+        const matches = /^@[^:]+:(.+)$/.exec(this.matrixClient.credentials.userId);
```

Comment:
Any benefit to this rather than just using a lazy matcher?

e.g `^@.+?:(.+)$`

Reply:
Probably not, this was just the first solution I thought of